main.tf

```diff
@@ -139,15 +139,11 @@ resource "aws_security_group" "this" {
   vpc_id      = "${var.vpc_id}"
 
   tags = "${var.tags}"
-}
-
-resource "aws_security_group_rule" "default_ingress" {
-  count = "${var.allowed_security_groups_count}"
 
-  type                     = "ingress"
-  from_port                = "${aws_rds_cluster.this.port}"
-  to_port                  = "${aws_rds_cluster.this.port}"
-  protocol                 = "tcp"
-  source_security_group_id = "${element(var.allowed_security_groups, count.index)}"
-  security_group_id        = "${aws_security_group.this.id}"
+  ingress {
+    from_port       = "${local.port}"
+    to_port         = "${local.port}"
+    protocol        = "tcp"
+    security_groups = "${var.allowed_security_groups}"
```

Comment:
I think is is a list: `security_groups = ["${var.allowed_security_groups}"]`

Reply:
if we make it like that. it will become a list within a list.

Reply:
> if we make it like that. it will become a list within a list 

Nope. It's just how TF works. e.g. https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/main.tf#L44

Reply:
oh thanks. didn't know that. but it's still working the way i have designed it. should i change it for consistency?